server/homebrew.api.js

```diff
@@ -37,14 +68,50 @@ const api = {
 		}
 		return { id, googleId };
 	},
+	getUsersBrewThemes : async (username, id)=>{
+		const fields = [
+			'title',
+			'tags',
+			'shareId',
+			'thumbnail',
+			'textBin',
+			'text',
+			'authors'
+		];
+
+		const userThemes = {
+			Brew : {
+
+			}
+		};
+
+		const brews = await HomebrewModel.getByUser(username, true, fields, { tags: { $in: ['meta:theme', 'meta:Theme'] }, shareId: { $ne: id }, renderer: { $ne: 'Legacy' } });
+
+		if(brews) {
+			for await (const brew of brews) {
+				userThemes.Brew[`#${brew.shareId}`] = {
```

Comment:
Can I get a brief overview of how the `#` plays into everything? I'm having trouble following what its purpose is.

Reply:
It is used to be a clear indicator that the theme is a user theme, so code doesn't try to load `\themes\V3\editId`

Reply:
Is there ever a case where such an endpoint would be generated? All static themes are going to be behind `\v3\themeName` and all user themes are going to be `\shareId`, right?

Reply:
There are places where themes are referenced by just the name (prop, front matter, etc.) with the expectation of being joined with the renderer field in the same structure vs places where they are not.  (IE `{renderer: 'V3', theme: '5ePHB'}` vs `/V3/53PHB`. 

I was also assuming that it was plausible for future rendering versions ( V4, etc. ) and that theme names could match with editID specs.

There were three obvious solutions to be certain a theme was a brew theme:

1. Add a "brew" value to the options for renderer - not future proof.
2. Add a new front-matter property/prop to indicate a brew theme 
3. "Tag" the theme name to indicate a brew theme in the places where themes are expressed in their string path form.
4. Walk the 'static' themes list to verify that the combination of renderer and theme was absent. Not horrible.

2 and 3 are the best choices as when HB is looking at the front matter metadata to load it currently only knows the renderer and a string that should represent a string. 4 isn't horrible.

I chose 3 because it is non-looping, adds no data structures, and is visually obvious.  I also wasn't always confident I'd have access to the global static themes structure. It doesn't take much to validate and acts as a smidge of a check bit. I don't think it would be hard to switch to 2 or 4, but I don't see a good enough reason to rework that little bit.

Reply:
Ok, I see where you're coming from now. We can keep it as-is for now.

I would eventually like to change his though; perhaps moving to option 2 or 4. The main reason being option 3 introduces an unique handling of brew IDs endpoints, where all the other endpoints (\share, \edit, \download, \source) use the ID directly without needing a `#` marker. It's a potential maintenance issue trying to remember that one endpoint has this different schema and remembering all the places that `#` is being handled.

I will note that `renderer` (Markdown version) and `theme` are separate entities. So 1. above wouldn't make sense anyway to select a brew as a renderer. As it stands now, *all* themes from user brews must use V3 renderer.

Reply:
While they are separate entities, there is no reason to expect that all Rendering Engine versions will have the same themes, they're pretty coupled.

If there is a preference for 2 or 4, let's just rip that bandaid now.

Reply:
> While they are separate entities, there is no reason to expect that all Rendering Engine versions will have the same themes, they're pretty coupled.

Exactly. But mainly, we don't want to overwrite the `renderer` field with a shareID because they are different values. Same reason we don't overwrite the `title` field with the `shareID`. They are two identifiers that serve different purposes.

> If there is a preference for 2 or 4, let's just rip that bandaid now.

I think I would prefer 4 for this. If `theme` is found in the list of static themes, load that. Otherwise assume it is a brew Id and try to load that.

Reply:
And done.

You'll also get to test the "the brew went away!" code if you have something saved with a brew theme. :)